### PR TITLE
9.1 free search

### PR DESCRIPTION
**Pull request by kimc0de**
**What we have done so far:**

- Hide search button for recruiter
- Show different alerts on the page (alert for first visit, alert for no match result, alert for missing job title)
- Add limited results dropdown (10, 20, 30, All results)
- Remain the input values after search
- Repeated search with different filters is possible

I also deleted the unused file that Ly mentioned, so that hopefully it wont make any merge conflict when Ly merges her branch. 

**Files that are deleted:**

- views/name.ejs
- views/candidates/index.ejs
- views/candidates/show.ejs
- public/javascripts/jobs/details.js

### CODE REVIEW

**Review by lyrated — APPROVED** [review]
some of the code in the search.ejs file could've been written a bit more compact, but this way is totally fine :D good job, everything also worked on my side

**Comment by kimc0de** [comment]
@lyrated hey which part u mean? u can give me example i can fix ^^

**Comment by lyrated** [comment]
> 
> 
> @lyrated hey which part u mean? u can give me example i can fix ^^

it's nothing that needs to be "fixed" since it's not wrong so don't worry :) I can still look up the lines of code I mean though

File: views/search/search.ejs [hunk 1]
```diff
@@ -1,61 +1,103 @@
...
+        if (size == undefined) { 
+          for ( var i = 0; i < options.length; i++ ){
+          %>
+          <option value="<%=options[i] %>"><%=result_labels[i] %> results</option>
+        <% }} else {
+        var options = [ "10", "20", "30", "10000" ];
+        var result_labels = [ "10", "20", "30", "All" ];
+        for ( var i = 0; i < options.length; i++ )
+        {
+            var selected = (size == options[i]) ? "selected" : "";
```
Comment (lyrated) [inline views/search/search.ejs:28]: here you declared the variables "options" and "result_labels" twice, and you also loop twice through the same arrays -> could've been shorter by just writing the 2nd loop and put the condition "typeof size !== 'undefined" inside this condition. so basically:
if size is not undefined and size is the same as options[i], then print selected. same thing with 42-63 ^^ but as I said, you don't have to worry about it

**Comment by kimc0de** [comment]
@lyrated ooh u totally right.. i forgot to delete the second var in else cuz at first we had let and then i pull them out as var.
And right the if else could have been inside the for loop <3
I just updated the code. If you have time pls check it to see if I understood what u meant

**Comment by lyrated** [comment]
> 
> I just updated the code. If you have time pls check it to see if I understood what u meant
Thanks for your hard work! Well it could be even shorter but basically this is what I meant, and it looks better now :D

**Review by lyrated — APPROVED** [review]

**Pull request merged**